modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.shared/src/main/scala/edu/gemini/seqexec/web/common/model.scala

```diff
@@ -7,20 +7,164 @@ import Scalaz._
   * Minimal models to be exchanged between client and server
   */
 case class StepConfig(key: String, value: String)
-case class Step(id: Int, config: List[StepConfig])
-case class SequenceSteps(steps: List[Step])
-case class Sequence(id: String, state: SequenceState, instrument: Instrument.Instrument, steps: SequenceSteps, error: Option[Int])
+
+sealed trait StepState
+
+/**
+  * Possible states for given step
+  */
+object StepState {
+  case object NotDone extends StepState
+  case object Running extends StepState
+  case object Done    extends StepState
+  case object Error   extends StepState
+  case object Abort   extends StepState
+  case object Paused  extends StepState
+
+  implicit val show = Show.shows[StepState] {
+    case Running    => "Running"
+    case NotDone    => "Pending"
```

Comment:
Right, I expect these objects to be properly defined on spmodel at some moment, for now they are mostly UI model classes